### PR TITLE
jk2338/add-socials-endpoint

### PR DESCRIPTION
**Pull request by JustKong13**
### Summary 

This pull request is the first step towards implementing the ability to add socials links to a club's document. After struggling to find a solution to check whether an URL is valid, I decided to create a new URL type URLs in types.ts to check whether the domain link is in the provided URL. Invalid URLs will throw an error. This endpoint would append a dictionary (consisting of a platform type and URL) to the array Socials in the clubtype. As of right now, I have req.params.id as the users's current ID to check if they are authenticated to change the club's file, until I figure out how to get the user's ID (Hopefully someone can help me with that :D)

- [x] Created and updated data types 
- [x] Created endpoint to add social media links to clubs
- [x] Make sure links are valid 

### Test Plan 

Tested locally with valid and invalid URLs, and different platforms. I have not tested the authentication check, but if there is a problem with this endpoint it is probably due to currentUser not being implemented or imported correctly

### Breaking Changes  

<!-- - Database schema change (anything that changes DynamoDB document structure)
<!-- - Other change that could cause problems (Detailed in notes)

### CODE REVIEW

File: backend/router/club.ts [hunk 1]
```diff
@@ -21,20 +21,43 @@ router.get('/', async (req, res) => {
...
+  const clubsDoc = clubsCollection.doc(req.body.id);
   console.log(req.body);
   const club: ClubType = req.body;
   await clubsDoc.set(club);
   res.send(club);
 });
 
+//Adds a club's social links (platform is other for custom social medias)
+router.post('/:id/socials/', async (req, res) => {
+  const clubID = req.body.id;
```
Comment (aaronkang7) [inline backend/router/club.ts]: This should use params and not req.body, as you are fetching the club id from the endpoint parameters (/:id) and not the body.

Comment (JustKong13) [reply backend/router/club.ts]: this should be fixed now!

**Comment by rgu0114** [comment]
I think you did a great job checking if club id's are valid and logging error messages to the console if they weren't. Didn't understand a lot of the code tbh but I think it looks solid tho. However, one part I was concerned about were the warning messages at the bottom of the changed files. Is there a way you can resolve them? I think a couple of them can be resolved by just streamlining the code.

**Comment by rgu0114** [comment]
> I think you did a great job checking if club id's are valid and logging error messages to the console if they weren't. Didn't understand a lot of the code tbh but I think it looks solid tho. However, one part I was concerned about were the warning messages at the bottom of the changed files. Is there a way you can resolve them? I think a couple of them can be resolved by just streamlining the code.

Like for example, u have several messages that say "*variable* defined but never used" and other redundant code (I think) u could get rid of or change so you do actually use them.

**Review by justinkang01 — APPROVED** [review]
code looks good to me

**Pull request merged**